pkg/cloud/vsphere/services/cloudprovider/addons.go

```diff
@@ -0,0 +1,277 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cloud-controller-manager",
+			Namespace: "kube-system",
+		},
+	}
+}
+
+// CloudControllerManagerService returns a Service for the cloud-controller-manager
+func CloudControllerManagerService() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cloud-controller-manager",
+			Namespace: "kube-system",
+			Labels: map[string]string{
+				"component": "cloud-controller-manager",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeNodePort,
+			Ports: []corev1.ServicePort{
+				{
+					Port:       443,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.FromInt(43001),
+				},
+			},
+			Selector: map[string]string{
+				"component": "cloud-controller-manager",
+			},
+		},
+	}
+}
+
+// CloudControllerManagerConfigMap returns a ConfigMap containing data for the cloud config file
+func CloudControllerManagerConfigMap(cloudConfig string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "vsphere-cloud-config",
+			Namespace: "kube-system",
+		},
+		Data: map[string]string{
+			"vsphere.conf": cloudConfig,
+		},
+	}
+}
+
+// CloudControllerManagerDaemonSet returns the DaemonSet which runs the cloud-controller-manager
+func CloudControllerManagerDaemonSet() *appsv1.DaemonSet {
+	return &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "vsphere-cloud-controller-manager",
+			Namespace: "kube-system",
+			Labels: map[string]string{
+				"k8s-app": "vsphere-cloud-controller-manager",
+			},
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"k8s-app": "vsphere-cloud-controller-manager",
+				},
+			},
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+				Type: appsv1.RollingUpdateDaemonSetStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"k8s-app": "vsphere-cloud-controller-manager",
+					},
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector: map[string]string{
+						"node-role.kubernetes.io/master": "",
+					},
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsUser: int64ptr(0),
+					},
+					Tolerations: []corev1.Toleration{
+						{
+							Key:    "node.cloudprovider.kubernetes.io/uninitialized",
+							Value:  "true",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "node-role.kubernetes.io/master",
+							Value:  "true",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+						{
+							Key:    "node.kubernetes.io/not-ready",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+					ServiceAccountName: "cloud-controller-manager",
+					Containers: []corev1.Container{
+						{
+							Name:  "vsphere-cloud-controller-manager",
+							Image: cloudProviderImage,
+							Args: []string{
+								"/bin/vsphere-cloud-controller-manager",
```

Comment:
The `/bin` part of this isn't necessary anymore because it's not party of the container image `ENTRYPOINT`.

The current examples have this removed as well: https://github.com/kubernetes/cloud-provider-vsphere/blob/551410442226d6e5d0754408917ead48c013c3eb/manifests/controller-manager/vsphere-cloud-controller-manager-pod.yaml#L22-L25

Reply:
Will update, thanks!